### PR TITLE
21744: Enhanced Trainee object to expose Trainee related operations directly, MAJOR

### PR DESCRIPTION
**Pull request by fulpm**
- Added Trainee operation methods to the Trainee class
- Fixed comments becoming all lowercase
- Updated Howso Engine to latest

### CODE REVIEW

**Comment by lancegliser** [comment]
The generated `react` code as an interesting comment.

```ts
  /**
   * Run reacts in a batch, output a an assoc of list of outputs from each individual react.
   * @param traineeId The Trainee identifier.
   * @param request The operation parameters.
   * @returns The response of the operation, including any warnings.
   */
  public async react(request: schemas.ReactRequest) {
    return this.client.react(this.id, request);
  }
```

"in a batch". We started separating our batch methods out just now. Wonder if there's potentially language alternatives we should use?

**Comment by lancegliser** [comment]
Will upgrade my implementation when `Trainee.delete()` comes in.

File: src/features/sources/Array.ts [hunk 1]
```diff
@@ -167,15 +160,12 @@ export class InferFeatureAttributesFromArray extends InferFeatureAttributesBase
...
-          (Array.isArray(options.tightBounds) && options.tightBounds.indexOf(featureName) === -1)
-        ) {
+        if (!options.tight_bounds) {
           // Use loose bounds
           [minValue, maxValue] = utils.guessLooseBounds(minValue, maxValue);
 
-          const { modeBounds = true } = options;
-          if (modeBounds || (Array.isArray(modeBounds) && modeBounds.indexOf(featureName) >= 0)) {
+          const { mode_bound_features = [] } = options;
+          if (mode_bound_features.includes(featureName)) {
```
Comment (fulpm) [inline src/features/sources/Array.ts:168]: mode bounds should still be checked if undefined

File: README.md [hunk 2]
```diff
@@ -85,15 +85,29 @@ import { type ClientOptions, HowsoWorkerClient, BrowserFileSystem } from "@howso
...
+  return await HowsoWorkerClient.create(worker, fs, {
     howsoUrl,
     migrationsUrl, // Optional, used for upgrading Trainees saved to disk.
     ...options,
   });
-  return client.setup();
 };
 ```
 
+Once you have a client you can then start creating your Trainees:
```
[the remark this answers is not in the document]
Comment (lancegliser) [reply README.md]: Added README directions for acquireTraineeResources

File: src/types/schemas/BuiltInFeatures.ts
[the remark this answers is not in the document]
Comment (lancegliser) [reply src/types/schemas/BuiltInFeatures.ts:1]: Types have been adjusted. We've dropped the more specific `${string}_delta_${number}` template literals out for now. Those will have to be maintained by end consumers at this time.

File: src/features/sources/Array.ts (hunk 1 above)
Comment (lancegliser) [reply src/features/sources/Array.ts]: Removed the defaulting of empty array, and using `!mode_bound_features || mode_bound_features.includes(featureName)` insatead.

**Review by apbassett — APPROVED** [review]

**Pull request merged**